aspnet/fundamentals/servers.rst

```diff
@@ -1,8 +1,274 @@
+
+Alternately, a configuration file can be referenced, instead:
+
+.. code-block:: javascript
+
+	"kestrel": "Microsoft.AspNet.Hosting --config hosting.ini"
+
+Then, ``hosting.ini`` can include the settings the server will use (including the server parameter, as well):
+
+.. code-block:: text
+
+	server=Kestrel
+	server.urls=http://localhost:5004
+
+Programmatic configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition to specifying configuration options, the server hosting the application can be referenced programmatically via the `IApplicationBuilder interface <https://github.com/aspnet/HttpAbstractions/blob/ed339a35d2a0ae6137c12e9bc8e8b037ed429bc1/src/Microsoft.AspNet.Http.Abstractions/IApplicationBuilder.cs>`_, available in the ``Configure`` method in ``Startup``. ``IApplicationBuilder`` exposes a Server property of type `ServerInformation <https://github.com/aspnet/Hosting/blob/62e534977386ebb02bfc351ec4a6bbea9ea7f635/src/Microsoft.AspNet.Hosting.Server.Abstractions/IServerInformation.cs>`_. ``IServerInformation`` only exposes a ``Name`` property, but different server implementations may expose additional functionality. For instance, WebListener exposes a ``Listener`` property that can be used to configure the server's authentication:
+
+.. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/Startup.cs
+	:linenos:
+	:lines: 29-45
+	:emphasize-lines: 3,6-8,13
+	:language: c#
+	:dedent: 8
+
+In addition to configuring the server, you can also adjust how your application will behave based on which server is hosting it, as provided by ``IServerInformation.Name`` (as shown on line 13 in the code listing above). The following table lists the servers and the name they provide:
+
+.. list-table::
+	:header-rows: 1
+	
+	* - Server
+	  - IServerInformation.Name
+	* - IIS and IIS Express
+	  - `Microsoft.AspNet.Server.IIS`
+	* - WebListener
+	  - `Microsoft.AspNet.Server.WebListener`
+	* - Kestrel
+	  - `Kestrel`
+
+	
+IIS and IIS Express
+-------------------
+
+The default web host for ASP.NET applications developed using Visual Studio 2015 is IIS / IIS Express. The "Microsoft.AspNet.Server.IIS" dependency is included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu.
+
+.. image:: /fundamentals/servers/_static/serverdemo-properties.png
+
+Working with IIS as your server for your ASP.NET application is a great option. It provides the most features, integrating with IIS and providing access to other IIS modules. It bypasses the legacy ``System.Web`` infrastructure used by prior versions of ASP.NET, providing a substantial performance gain. IIS has great support for static files and can also be used with the built-in Windows Authentication mechanism, too.
```

Comment:
The tone of this paragraph is out of sync with the rest of the page. This sounds like opinion and advertising rather than documentation.